test/unit/question_base_test.rb

```diff
@@ -90,6 +90,19 @@ class QuestionBaseTest < ActiveSupport::TestCase
         assert_equal [:done], @question.permitted_next_nodes
       end
     end
+
+    context 'next_node called with block but not permitted option specified' do
+      should 'return nodes returned via syntactic sugar methods' do
+        @question.next_node do |response|
+          if response == 'yes'
+            outcome :done
+          else
+            question :another_question
+          end
+        end
+        assert_equal [:done, :another_question], @question.permitted_next_nodes
+      end
+    end
   end
```

Comment:
Regarding your second comment - I've tried to improve the test name. Fixup commit: 7c9ea426dd2e1dadda4607db5e74cb7908ab27af.